### PR TITLE
Add CI testing for nitro

### PR DESCRIPTION
**Pull request by a-thomas-22**
Adds CI tests for charts using [ct](https://github.com/helm/chart-testing) on a [kind cluster](https://kind.sigs.k8s.io/)

The files in each charts `/tests` directory define the tests. The files in each chart's `/ci` directory define the values that test will be run against.

There are two new workflows that will run. One installs the charts and one runs an upgrade against the previous chart semver. The upgrade test will fail until this is merged since the last version did not support tests.

The tests for the `das` and `relay` charts simply ensure the pods mark themselves as ready with the values provided in the `/ci` folder since there are currently no tests. Unit tests can be added in the future.

The nitro chart includes a test image which runs unit tests. The PR for this application is https://github.com/OffchainLabs/sre-applications/pull/53 . The intent is not to replicate nitro unit tests, but to make sure nitro is working correctly within the scope of the helm chart(does the pod become ready, are metrics available if enabled, does the rpc work on the port we expect, etc...). After merging this initial PR the tests will be expanded and test images will be created for the das and the relay.

This PR also includes a fix for improperly nested yaml in the nitro chart.

### CODE REVIEW

File: charts/nitro/templates/tests/test.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: "{{ include "nitro.fullname" . }}-ci-configmap"
+data:
+  config.yaml: |
+    {{- toYaml .Values.ci | nindent 4 }}
```
Comment (lambchr) [inline charts/nitro/templates/tests/test.yaml:7]: does this mean the `ci.secrets` will also be present in this config map?

File: charts/nitro/templates/tests/test.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    {{- include "nitro.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": test
+spec:
+  containers:
+  - name: sre-nitro-test-suite
+    image: public.ecr.aws/f6s7v9z5/offchain-labs/sre-nitro-test:v0.0.6
+    {{- if .Values.ci.debug.enabled }}
+    command: ["/bin/sh", "-c", "--"]
+    args: ["while true; do sleep 30; done;"]
```
Comment (lambchr) [inline charts/nitro/templates/tests/test.yaml:34]: shouldn't we be running the test tool from the image rather than sleeping?

**Review by lambchr — COMMENTED** [review]
nice, this will be a great addition! I just have a couple of comments :)

File: .github/workflows/test-install.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            # Append to the secrets string in the required format
+            if [ -z "$secrets_for_helm" ]; then
+              secrets_for_helm="ci.secrets.${variable_name}=${secret_value}"
+            else
+              secrets_for_helm="${secrets_for_helm},ci.secrets.${variable_name}=${secret_value}"
+            fi
+          done
+          
+          # Remove the initial comma to clean up the format
+          secrets_for_helm=${secrets_for_helm#,}
```
[the remark this answers is not in the document]
Comment (a-thomas-22) [reply .github/workflows/test-install.yaml:63]: While the if statement attempts to handle the format correctly, the way the loop is structured still results in a leading comma for the first iteration. This line is indeed necessary to clean up the format and remove the leading comma.

File: .github/workflows/test-upgrade.yaml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+          # Use the command directly or set the HELM_EXTRA_SET_ARGS environment variable for later use in the workflow
+          echo "HELM_EXTRA_SET_ARGS=${helm_extra_set_args}" >> $GITHUB_ENV
+        
+        
+      - name: Run chart-testing (install)
+        if: steps.list-changed.outputs.changed == 'true'
+        run: |
+          ct install --target-branch ${{ github.event.repository.default_branch }} --config ct.yaml --upgrade --helm-extra-set-args "$HELM_EXTRA_SET_ARGS"
+
```
[the remark this answers is not in the document]
Comment (a-thomas-22) [reply .github/workflows/test-upgrade.yaml]: The `--upgrade` flag is there, but I fixed the step name to be more descriptive

File: charts/nitro/templates/tests/test.yaml (hunk 1 above)
Comment (a-thomas-22) [reply charts/nitro/templates/tests/test.yaml:7]: great catch.

File: charts/nitro/templates/tests/test.yaml (hunk 2 above)
Comment (a-thomas-22) [reply charts/nitro/templates/tests/test.yaml:24]: Thats only run if `.Values.ci.debug.enabled ` is true. Making sure the secrets were loaded correctly required using this feature extensively 🙃

Comment (lambchr) [reply charts/nitro/templates/tests/test.yaml:24]: ah whoops, sorry I missed that part, thanks for explaining :)

File: .github/workflows/test-upgrade.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (lambchr) [reply .github/workflows/test-upgrade.yaml]: whoops, missed that, sorry!

File: charts/nitro/templates/tests/test.yaml (hunk 1 above)
Comment (lambchr) [reply charts/nitro/templates/tests/test.yaml]: You asked for a re-review but as far as I can see it still looks like the secrets would be added to this config map. Did you want to change that? Let me know if I am mistaken :)

**Review by lambchr — COMMENTED** [review]
generally lgtm, just one question still remaining on the config map

**Review by lambchr — APPROVED** [review]
lgtm! My last review I was looking at a previous commit and it didn't load the ConfigMap changes that were added :)

**Pull request merged**